CMakeLists.txt

```diff
@@ -201,49 +172,58 @@ endif()
 
 if(UNIX)
     # On unix-like platforms the library is almost always called libz
-   set_target_properties(zlib zlibstatic PROPERTIES OUTPUT_NAME z)
-   if(NOT APPLE)
-     set_target_properties(zlib PROPERTIES LINK_FLAGS "-Wl,--version-script,\"${CMAKE_CURRENT_SOURCE_DIR}/zlib.map\"")
-   endif()
+    set_target_properties(zlib PROPERTIES OUTPUT_NAME z)
+    if(BUILD_SHARED_LIBS AND NOT APPLE)
+        set_target_properties(zlib PROPERTIES LINK_FLAGS "-Wl,--version-script,\"${CMAKE_CURRENT_SOURCE_DIR}/zlib.map\"")
+    endif()
 elseif(BUILD_SHARED_LIBS AND WIN32)
     # Creates zlib1.dll when building shared library version
     set_target_properties(zlib PROPERTIES SUFFIX "1.dll")
 endif()
 
-if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL )
-    install(TARGETS zlib zlibstatic
+if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
+    install(TARGETS zlib
+        EXPORT ${PROJECT_NAME}Config
         RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
         ARCHIVE DESTINATION "${INSTALL_LIB_DIR}"
-        LIBRARY DESTINATION "${INSTALL_LIB_DIR}" )
+        LIBRARY DESTINATION "${INSTALL_LIB_DIR}")
+    install(EXPORT ${PROJECT_NAME}Config
+        NAMESPACE zlib:: DESTINATION "${INSTALL_LIB_DIR}/cmake")
```

Comment:
If installing to a namespace like `zlib::` (a good idea in general), it would be helpful to also define an alias target for those projects incorporating the zlib repo directly via `add_subdirectory()`. This is becoming much more commonplace with the `FetchContent` module added in CMake 3.11. By adding a line like the following, other projects can build against an installed zlib or by incorporating zlib directly into their own build via `add_subdirectory()`.

```cmake
add_library(zlib::zlib ALIAS zlib)
```

Note also that CMake's own `FindZLIB.cmake` module defines a slightly different imported target named `ZLIB::ZLIB` (the upper/lowercase differences are meaningful). It might therefore also be wise to define this alias as well for similar reasons, leading to:

```cmake
add_library(zlib::zlib ALIAS zlib)
add_library(ZLIB::ZLIB ALIAS zlib)
```

Reply:
You also want to install the `${PROJECT_NAME}Config` file to `${INSTALL_LIB_DIR}/cmake/${PROJECT_NAME}` if you want CMake to be able to find it automatically (assuming the base install directory is part of the search path, such as being specified in `CMAKE_PREFIX_PATH` or with zlib installed to somewhere like `/usr` on Unix).

Reply:
done, thanks for the note